### ISSUE TITLE
[BUG] The applicate has the “Direct-to-Consumer Delivery”, but the RDT request return “Access to requested resource is denied.”

### ISSUE THREAD

**Issue by NancyLin** [amzn/selling-partner-api-docs#1541]
Hello,

Our application has the “Direct-to-Consumer Delivery” role, and we use the Self authorization, but the RDT request return “Access to requested resource is denied.” error.

But we requested the RDT API successfully before. and last night, we were suddenly prompted with an "Access to requested resource is denied" error.

What should we do? please help us. now we can not deliver orders without the consumer's address.

![image](https://user-images.githubusercontent.com/8305040/138024241-a5a323f7-f583-4c49-912f-3e601293b3a1.png)

RDT Request Body:
{
    "method":"GET",
    "path":"/orders/v0/orders",
    "dataElements":[
        "buyerInfo",
        "shippingAddress"
    ]
}

RDT response:
{
  "errors": [
    {
      "message": "Access to requested resource is denied.",
     "code": "Unauthorized",
     "details": ""
    }
  ]
}

**Comment by NancyLin** [comment]
> Yes, but it is normal when not use RDT

Yes, but we need the address to deliver orders, and our application has the "Direct-to-Consumer Delivery" role.

**Comment by NancyLin** [comment]
@diegocvazquez  No, I have created a case but not receive a reply today.

**Comment by diegocvazquez** [comment]
@l-o-u-k-e 
are you using a user or a role ARN in your application?

**Comment by l-o-u-k-e** [comment]
@diegocvazquez I'm using a role, as suggested in the SP-API setup reference

**Comment by l-o-u-k-e** [comment]
@diegocvazquez I'm sorry but as far as I can see, now I can create RDT as usual, nothing today was changed in my source code.
[buyerInfo, shippingAddress] are in fact PII data, and I'm still receiving them.
But I must specify that I'm querying GET_ORDER_REPORT_DATA_INVOICING report and, as my company receive 1 Amazon order per day (or less), my process is scheduled to download same past days' orders list each hour.
I hope that when I will receive new order(s), everything will be ok.

**Comment by lemnisk8** [comment]
Issue seems resolved here... getOrder & getOrders  - EU endpoint

**Comment by anghnam** [comment]
Seems like the issue has been resolved for EU region. Works fine now

**Comment by DomasWEB** [comment]
Hi @DawidWawrzyniak , maybe you can help me with this role? :) LinkedIn: https://www.linkedin.com/in/dominykastijunaitis/

**Comment by dnErf** [comment]
Hi, I am having same issue as well, I can get our RDT but when i used it on orders endpoint and replacing LWA token to RDT im getting denied.

Is there any update on this issue? or is there a work around to get the shipping address? Thanks for any help.

**Comment by dingledorf** [comment]
@dnErf . I'm getting the exact same issue. Were you able to resolve? Thanks.

**Comment by dnErf** [comment]
@dingledorf . Hello, unfortunately no. We have other ongoing project so this one was put on hold but I'm guessing that it was because of how store authorization settings link up with the application.

**Comment by daniel031699** [comment]
@dingledorf @dnErf Now I can get the RDT correctly. When calling the getorders interface with RDT, I always prompt unauthorized. Can you help me?

**Comment by younkim** [comment]
Did any of you manually request Amazon to get permissions on the restricted roles as described here? https://developer-docs.amazon.com/sp-api/docs/roles-in-the-selling-partner-api

I'm also getting 403 despite following everything, and everything else other than the PII request itself not working. I found the text "request and quality..." in that link that leads me to believe that maybe we need to request for access outside of Developer Central. The dev page has delegated app restricted permission settings, but not for your own private app.

**Comment by AlbertoCamachoMunoz** [comment]
Yesterday I was making API calls using postman, all perfect response, correct results until suddenly ERROR
Without having changed absolutely anything
I'm using user iam with permision for access to all marketplace, 
when i 'm using the api for get autorization response with token the  Atza|XXXX, i use this token for get Order response until the last night when the aplication get error in the response

**Comment by AmandeepSingh179** [comment]
@Mike-the-one were you able to get it working. I'm facing the same issue.

**Comment by densai09** [comment]
same error, when request RDT. {
  "errors": [
    {
      "code": "InvalidInput",
      "message": "Application does not have access to one or more requested data elements: [buyerInfo, shippingAddress]",
      "details": ""
    }
  ]
}

**Comment by martr84** [comment]
We had tried to get this working we went through the process of clicking edit app and ticking the DirectToConsumer Role after being approved for the role and then regenerating the refresh token. however that DID NOT work.

What DID work is creating a brand new app registration with the same role arn and ticking all the required roles as part of the initial setup. This gave us new lwa credentials, new app ids and new refresh tokens.

Hopefully this saves many hours of frustrations for other people.
